kimchi/src/circuits/polynomials/foreign_field_mul/witness.rs

```diff
@@ -164,6 +148,14 @@ pub fn create<F: PrimeField>(
     right_input: &BigUint,
     foreign_field_modulus: &BigUint,
 ) -> ([Vec<F>; COLUMNS], ExternalChecks<F>) {
+    if *foreign_field_modulus > BigUint::max_foreign_field_modulus::<F>() {
```

Comment:
Similarly inefficient; similarly shouldn't be in this PR

Reply:
Just to be clear, this was already in the master branch. Meaning, it is not added in this `ffmul/new` branch for the first time. But sure, let's keep it out. If you know another more efficient way to check that we are not exceeding the maximum value, let me know so I can add another PR accordingly.